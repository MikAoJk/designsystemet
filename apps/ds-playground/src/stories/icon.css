.myIconClassname {
  fill: var(--palette-denim-100);
}
