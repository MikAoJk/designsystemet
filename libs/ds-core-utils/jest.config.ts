/* eslint-disable */
export default {
  displayName: 'ds-core-utils',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nrwl/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/ds-core-utils',
};
