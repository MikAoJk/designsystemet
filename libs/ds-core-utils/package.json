{
  "name": "@skatteetaten/ds-core-utils",
  "groupId": "no.skatteetaten.aurora",
  "version": "0.0.1",
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "react": ">=17"
  }
}
