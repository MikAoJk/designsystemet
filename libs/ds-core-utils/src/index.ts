export * from './SymbolLogo/SymbolLogo';
export * from './base-props.types';
export * from './defaults';
export * from './example-layouts/ExternalLayout';
export * from './example-layouts/InternalLayout';
export * from './i18n';
export * from './Separator/Separator';
