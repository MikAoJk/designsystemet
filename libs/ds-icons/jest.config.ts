/* eslint-disable */
export default {
  displayName: 'ds-icons',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nx/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/ds-icons',
};
