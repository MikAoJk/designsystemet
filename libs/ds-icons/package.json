{
  "name": "@skatteetaten/ds-icons",
  "groupId": "no.skatteetaten.aurora",
  "version": "0.5.0",
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^13",
    "@skatteetaten/ds-core-utils": "*"
  },
  "scripts": {}
}
