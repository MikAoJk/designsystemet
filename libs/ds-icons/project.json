{
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-icons/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "build": {
      "executor": "@nrwl/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-icons",
        "tsConfig": "libs/ds-icons/tsconfig.lib.json",
        "project": "libs/ds-icons/package.json",
        "entryFile": "libs/ds-icons/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-icons/rollup.config.js"],
        "compiler": "babel",
        "assets": [
          {
            "glob": "libs/ds-icons/README.md",
            "input": ".",
            "output": "."
          }
        ]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/ds-icons/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/ds-icons"],
      "options": {
        "jestConfig": "libs/ds-icons/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-icons/**/*.css",
          "libs/ds-icons/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      }
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-icons"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish {args.switches}",
        "cwd": "dist/libs/ds-icons"
      }
    }
  }
}
