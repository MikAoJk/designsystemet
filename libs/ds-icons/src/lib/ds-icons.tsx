import styles from './ds-icons.module.scss';

export function DsIcons(): JSX.Element {
  return (
    <div className={styles['container']}>
      <h1>{'Welcome to DsIcons!'}</h1>
    </div>
  );
}

export default DsIcons;
