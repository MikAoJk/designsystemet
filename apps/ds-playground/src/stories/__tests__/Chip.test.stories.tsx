import { sizeArr } from '@skatteetaten/ds-core-utils';
import { WarningSVGpath } from '@skatteetaten/ds-icons';
import { Chip, chipVariants } from '@skatteetaten/ds-status';
import { expect } from '@storybook/jest';
import { Meta, StoryObj } from '@storybook/react';
import { within } from '@storybook/testing-library';

import { SystemSVGPaths } from '../utils/icon.systems';

const meta = {
  component: Chip,
  title: 'Tester/Chip',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: {
      table: { disable: false },
      control: 'text',
    },
    svgPath: {
      table: { disable: true },
      options: Object.keys(SystemSVGPaths),
      mapping: SystemSVGPaths,
    },
    size: {
      options: [sizeArr[1], sizeArr[2]],
      control: 'radio',
      table: { disable: true },
    },
    variant: {
      options: chipVariants,
      control: 'radio',
      table: { disable: true },
    },
  },
} satisfies Meta<typeof Chip>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultText = 'SÃ¦ravgift';

const defaultArgs = {
  children: defaultText,
};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLDivElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    await expect(container).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5)',
  args: {
    ...defaultArgs,
    id: 'htmlId',
    className: 'dummyClassname',
    lang: 'en',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    await expect(container).toHaveAttribute('id', 'htmlId');
    await expect(container).toHaveClass('dummyClassname');
    await expect(container).toHaveAttribute('lang', 'en');
    await expect(container).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const DefaultYellow = {
  name: 'Default Variant (yellow) (A3)',
  args: {
    ...defaultArgs,
    variant: 'yellow',
  },
  argTypes: {
    variant: {
      table: { disable: true },
    },
  },
} satisfies Story;

export const DefaultGreen = {
  name: 'Variant (green) (A3)',
  args: {
    ...defaultArgs,
    variant: 'green',
  },
  argTypes: {
    variant: {
      table: { disable: true },
    },
  },
} satisfies Story;

export const DefaultRed = {
  name: 'Variant (red) (A3)',
  args: {
    ...defaultArgs,
    variant: 'red',
  },
  argTypes: {
    variant: {
      table: { disable: true },
    },
  },
} satisfies Story;

export const DefaultGrey = {
  name: 'Variant (grey) (A3)',
  args: {
    ...defaultArgs,
    variant: 'grey',
  },
  argTypes: {
    variant: {
      table: { disable: true },
    },
  },
} satisfies Story;

export const Small = {
  name: 'Small (A1)',
  args: {
    ...defaultArgs,
    size: 'small',
  },
  argTypes: {
    size: {
      table: { disable: true },
    },
  },
} satisfies Story;

export const WithCustomIcon = {
  name: 'With Custom Icon (A5)',
  args: {
    ...defaultArgs,
    svgPath: WarningSVGpath,
    variant: 'red',
  },
  argTypes: {
    svgPath: {
      table: { disable: false },
    },
  },
} satisfies Story;
