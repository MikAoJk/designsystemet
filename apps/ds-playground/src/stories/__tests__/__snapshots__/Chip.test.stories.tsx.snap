// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Chip DefaultGreen smoke-test 1`] = `
<div data-test-block="true">
  <div class="Chip_chip__L45zR Chip_chip_green__tDtT+ Chip_chip_medium__etsgc ">
    Særavgift
  </div>
</div>
`;

exports[`Tester/Chip DefaultGrey smoke-test 1`] = `
<div data-test-block="true">
  <div class="Chip_chip__L45zR Chip_chip_grey__UY0lo Chip_chip_medium__etsgc ">
    Særavgift
  </div>
</div>
`;

exports[`Tester/Chip DefaultRed smoke-test 1`] = `
<div data-test-block="true">
  <div class="Chip_chip__L45zR Chip_chip_red__BB3tt Chip_chip_medium__etsgc ">
    Særavgift
  </div>
</div>
`;

exports[`Tester/Chip DefaultYellow smoke-test 1`] = `
<div data-test-block="true">
  <div class="Chip_chip__L45zR Chip_chip_yellow__xgxPh Chip_chip_medium__etsgc ">
    Særavgift
  </div>
</div>
`;

exports[`Tester/Chip Small smoke-test 1`] = `
<div data-test-block="true">
  <div class="Chip_chip__L45zR Chip_chip_yellow__xgxPh Chip_chip_small__mtn+q ">
    Særavgift
  </div>
</div>
`;

exports[`Tester/Chip WithAttributes play-test 1`] = `
<div data-test-block="true">
  <div id="htmlId"
       lang="en"
       data-testid="123ID"
       class="Chip_chip__L45zR Chip_chip_yellow__xgxPh Chip_chip_medium__etsgc dummyClassname"
  >
    Særavgift
  </div>
</div>
`;

exports[`Tester/Chip WithCustomIcon smoke-test 1`] = `
<div data-test-block="true">
  <div class="Chip_chip__L45zR Chip_chip_red__BB3tt Chip_chip_medium__etsgc ">
    <div class="Chip_iconWrapper__KjHgM"
         aria-hidden="true"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_small__Yzxmo "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
        </path>
      </svg>
    </div>
    Særavgift
  </div>
</div>
`;

exports[`Tester/Chip WithRef play-test 1`] = `
<div data-test-block="true">
  <div class="Chip_chip__L45zR Chip_chip_yellow__xgxPh Chip_chip_medium__etsgc "
       id="dummyIdForwardedFromRef"
  >
    Særavgift
  </div>
</div>
`;
